datalad/distribution/add_sibling.py

```diff
@@ -0,0 +1,183 @@
+            args=('sibling',),
+            doc="""Name of the sibling to be added. If RECURSIVE is set, the
+                same name will be used to address the subdatasets' siblings""",
+            constraints=EnsureStr() | EnsureNone()),
+        url=Parameter(
+            args=('url',),
+            doc="""The URL of or path to the dataset sibling named by
+                `sibling`.
+                If you want to recursively add siblings, it is expected, that
+                you pass a template for building the URLs of the siblings of
+                all (sub)datasets by using placeholders.\n
+                List of currently available placeholders:\n
+                %%NAME\tthe name of the dataset, where slashes are replaced by
+                dashes.\nThis option is ignored if there is already a
+                configured sibling dataset under the name given by `sibling`""",
+            constraints=EnsureStr() | EnsureNone()),
+        pushurl=Parameter(
+            args=('--pushurl',),
+            doc="""In case the `url` cannot be used to publish to the dataset
+                sibling, this option specifies a URL to be used instead.\n
+                This option is ignored if there is already a configured sibling
+                dataset under the name given by `sibling`""",
+            constraints=EnsureStr() | EnsureNone()),
+        recursive=Parameter(
+            args=("--recursive", "-r"),
+            action="store_true",
+            doc="""Recursively add the `sibling` to all subdatasets of
+                `dataset`""",),
+        force=Parameter(
+            args=("--force", "-f",),
+            action="store_true",
+            doc="""If `sibling` exists already, force to (re-)configure its
+                URLs""",),)
+
+    @staticmethod
+    @datasetmethod(name='add_sibling')
+    def __call__(dataset=None, sibling=None, url=None,
+                 pushurl=None, recursive=False, force=False):
+
+        # TODO: Detect malformed URL and fail?
+
+        if sibling is None or url is None:
+            raise ValueError("""insufficient information to add a sibling
+                (needs at least a dataset, a name and an URL).""")
+
+        # shortcut
+        ds = dataset
+
+        if ds is not None and not isinstance(ds, Dataset):
+            ds = Dataset(ds)
+        if ds is None:
+            # try to find a dataset at or above CWD
+            dspath = GitRepo.get_toppath(abspath(getpwd()))
+            if dspath is None:
+                raise ValueError("""No dataset found
+                                 at or above {0}.""".format(getpwd()))
+            ds = Dataset(dspath)
+            lgr.debug("Resolved dataset for target creation: {0}".format(ds))
+
+        assert(ds is not None and sibling is not None and url is not None)
+
+        if not ds.is_installed():
+            raise ValueError("""Dataset {0} is not installed yet.""".format(ds))
+        assert(ds.repo is not None)
+
```

Comment:
Previous 20 lines are copied verbatim from create_publication_target...py -- looks like a function wanting to be born.